### PR TITLE
Report unhandled exceptions

### PR DESCRIPTION
**Pull request by digeff**
Now we report unhandled exceptions and unhandled rejections in telemetry

### CODE REVIEW

File: src/adapter/debugAdapter.ts [hunk 1]
```diff
@@ -106,6 +106,9 @@ export class DebugAdapter {
 
   async _onSetBreakpoints(params: Dap.SetBreakpointsParams): Promise<Dap.SetBreakpointsResult | Dap.Error> {
     const ids = generateBreakpointIds(params);
+    setTimeout(() => {
+       throw new Error('aweffeaw');
+    }, 10000);
```
Comment (connor4312) [inline src/adapter/debugAdapter.ts]: 🔥

File: src/telemetry/unhandledErrorReporter.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { extractErrorDetails, RawTelemetryReporter } from './telemetryReporter';
+
+type ExceptionType = 'uncaughtException' | 'unhandledRejection';
+
+export function installUnhandledErrorReporter(telemetryReporter: RawTelemetryReporter): void {
+  process.addListener('uncaughtException', (exception: unknown) => {
```
Comment (connor4312) [inline src/telemetry/unhandledErrorReporter.ts]: I believe this is run in the extension host within vscode, so we may end up getting exceptions thrown by other exceptions here. We probably want to filter these out, perhaps by checking its stacktrace to ensure it's coming from one of our scripts?

File: src/adapter/debugAdapter.ts (hunk 1 above)
Comment (digeff) [reply src/adapter/debugAdapter.ts]: Thanks for catching this... This is how I was testing the code xD

File: src/telemetry/unhandledErrorReporter.ts (hunk 2 above)
Comment (roblourens) [reply src/telemetry/unhandledErrorReporter.ts]: I am also still not sure whether we should be moving it out of the EH to run in a separate process like other debug adapters.

Comment (digeff) [reply src/telemetry/unhandledErrorReporter.ts]: Done

File: src/adapter/debugAdapter.ts (hunk 1 above)
Comment (digeff) [reply src/adapter/debugAdapter.ts]: Done

File: src/telemetry/unhandledErrorReporter.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  return errMsg;
+}
+
+const debugAdapterFolder = path.dirname(path.dirname(path.dirname(__dirname)));
+
+function shouldReportThisError(error: any): boolean {
+  // In VS Code, this debug adapter runs inside the extension host process, so we could capture
+  // errors from other pieces of software here. We check to make sure this is our error before reporting it
+  return !shouldFilterErrorsReportedToTelemetry
+    || (error && typeof error.stack === 'string' && error.stack.indexOf(debugAdapterFolder) !== -1);
```
Comment (connor4312) [inline src/telemetry/unhandledErrorReporter.ts]:
```suggestion
    || (error && typeof error.stack === 'string' && error.stack.includes(debugAdapterFolder));
```

You can use the ES6 `String.prototype.includes`

**Pull request merged**